File: C++/product-of-array-except-self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int sz = nums.size();
        vector<int> ans;
        long prod = 1;
        for (int j = 0; j < sz; j++) {
            prod *= nums[j];    
        }
        for (int i = 0; i < sz; i++) {
            if (nums[i] == 0) {
                int p = 1;
                for (int j = 0; j < sz; j++) {
                    if (i != j)
                        p *= nums[j];    
                }
                ans.push_back(p);
            } else {
                ans.push_back((int)(prod / nums[i]));    
            }         
        }
        return ans;
    }
};

File: Go/product-of-array-except-self.go
func productExceptSelf(nums []int) []int {
	n := len(nums)
	result := make([]int, n)
	prod := 1
	for i := 0; i < n; i++ {
        result[i] = prod
        prod *= nums[i]
	}
	prod = 1
	for i := n - 1; i >= 0; i-- {
		result[i] *= prod
        prod *= nums[i]
	}
	return result
}